File: sort_b1b2b3.py
def sort_b1b2b3(b1b2b3):
    ind = []
    for b in b1b2b3:
        indices = [b.find('W'), b.find('Y'), b.find('R')]
        sorted_indices = sorted(range(len(indices)), key=lambda k: indices[k])
        ind.append(sorted_indices)
    return ind

File: sort_b1b2b3.m
function ind = sort_b1b2b3(b1b2b3)
ind = zeros(length(b1b2b3),3);
for i = 1:length(b1b2b3)
   [b123, ind(i,:)] = sort([strfind(b1b2b3{i},'W') strfind(b1b2b3{i},'Y') strfind(b1b2b3{i},'R')]);
end